src/Microsoft.AspNetCore.Razor.Evolution/DirectiveDescriptorType.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public enum DirectiveDescriptorType
```

Comment:
IMO we should suffix enums like this with `Kind` instead of `Type`. It's easy to overload the meaning of type.